datalake/src/main/java/com/sequenceiq/datalake/service/sdx/CloudStorageManifester.java

```diff
@@ -60,23 +65,63 @@ private void setStorageLocations(FileSystemParameterV4Responses fileSystemRecomm
         cloudStorageRequest.setLocations(storageLocations);
     }
 
-    private void setIdentities(SdxCloudStorageRequest cloudStorage, CloudStorageRequest cloudStorageRequest) {
-        StorageIdentityBase logStorageIdentity = new StorageIdentityBase();
-        logStorageIdentity.setType(CloudIdentityType.LOG);
+    private void setStorageParameters(DetailedEnvironmentResponse environment, CloudStorageRequest cloudStorageRequest) {
+        if (isS3GuardConfigured(environment)) {
+            if (!Strings.isNullOrEmpty(environment.getAws().getS3guard().getDynamoDbTableName())) {
+                AwsStorageParameters awsStorageParameters = new AwsStorageParameters();
+                S3Guard s3Guard = new S3Guard();
+                s3Guard.setDynamoTableName(environment.getAws().getS3guard().getDynamoDbTableName());
+                awsStorageParameters.setS3Guard(s3Guard);
+                cloudStorageRequest.setAws(awsStorageParameters);
+            }
+        }
+    }
+
+    private boolean isS3GuardConfigured(DetailedEnvironmentResponse environment) {
+        return environment.getAws() != null && environment.getAws().getS3guard() != null;
```

Comment:
I think one more check could be useful here: `&& StringUtils.isNotEmpty(environment.getAws().getS3guard(). getDynamoDbTableName ())`

In case for some reason there is an S3Guard object with a null or empty dynamo table name. But I'm not sure.

Reply:
Oh I thought if s3guard is presented then dynametable is required

Reply:
anyway I think this what you searching https://github.com/hortonworks/cloudbreak/pull/5799/files#diff-932cffad5e6885b23e7012f1ca3a3085R70